tests/unit/h/conftest.py

```diff
@@ -146,7 +146,14 @@ def pyramid_request(db_session, db_session_replica, fake_feature, pyramid_settin
     )
     request.default_authority = "example.com"
     request.create_form = mock.Mock()
-    request.matched_route = mock.Mock()
+
+    request.matched_route = mock.Mock(spec=["name"])
+    # `name` is a special argument to the Mock constructor so if you want a
+    # mock to have a `name` attribute you can't just do Mock(name="name"), you
+    # have to use a separate call to configure_mock() instead.
+    # https://docs.python.org/3/library/unittest.mock.html#mock-names-and-the-name-attribute
+    request.matched_route.configure_mock(name="index")
```

Comment:
Previously `request.matched_route` was just an unconstrained mock in the tests, meaning `request.matched_route.name` was also an unconstrained mock, so `requests.matched_route.name.startswith("anything")` would return a mock which would be truthy. So `_is_api_request()` would consider all requests to be API requests.

Fixing this to be a spec'd mock with an actual string for `name`.